code/triangle/ViewModel/src/test/java/ru/unn/agile/triangle/viewmodel/ViewModelTests.java

```diff
@@ -0,0 +1,372 @@
+package ru.unn.agile.triangle.viewmodel;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import ru.unn.agile.triangle.Triangle.*;
+
+import static org.junit.Assert.*;
+
+public class ViewModelTests {
+    private ViewModel viewModel;
+
+    @Before
+    public void setUp() {
+        viewModel = new ViewModel();
+    }
+
+    @After
+    public void tearDown() {
+        viewModel = null;
+    }
+
+    @Test
+    public void canSetDefaultValues() {
+        assertEquals("", viewModel.axProperty().get());
+        assertEquals("", viewModel.ayProperty().get());
+        assertEquals("", viewModel.bxProperty().get());
+        assertEquals("", viewModel.byProperty().get());
+        assertEquals("", viewModel.cxProperty().get());
+        assertEquals("", viewModel.cyProperty().get());
+        assertEquals(Operation.GETANGLEA, viewModel.operationProperty().get());
+        assertEquals("", viewModel.resultProperty().get());
+        assertEquals(Status.WAITING.toString(), viewModel.statusProperty().get());
+    }
+
+    @Test
+    public void statusIsWaitingWhenCalculateWithEmptyFields() {
+        viewModel.calculate();
+        assertEquals(Status.WAITING.toString(), viewModel.statusProperty().get());
+    }
+
+    @Test
+    public void statusIsReadyWhenFieldsAreFill() {
+        viewModel.axProperty().set("1");
+        viewModel.ayProperty().set("2");
+        viewModel.bxProperty().set("3");
+        viewModel.byProperty().set("4");
+        viewModel.cxProperty().set("5");
+        viewModel.cyProperty().set("6");
+
+        assertEquals(Status.READY.toString(), viewModel.statusProperty().get());
+    }
+
+    @Test
+    public void canReportBadFormat() {
+        viewModel.axProperty().set("a");
```

Comment:
Is it correct if user sets  value=" 2 "  (with whitespaces)? Add test.

Reply:
but it is correct for my API.